packages/sodium/lib/src/api/secure_key_extensions.dart

```diff
@@ -0,0 +1,52 @@
+import 'secure_key.dart';
+import 'sodium.dart';
+
+extension SecureKeySplit on SecureKey {
+  /// Creates multiple secure keys of different [lengths] from a single key.
+  /// It is especially useful for using pwHash to generate enough bytes for
+  /// multiple keys.
+  ///
+  /// The returned keys are independent copies of portions of the original key,
+  /// with the same kind of memory protection. The copying is done on secure,
+  /// native memory controlled by [sodium], without exposing the data to dart
+  /// when using the dart VM.
+  List<SecureKey> split(Sodium sodium, List<int> lengths) => runUnlockedSync(
+        (originalKeyData) {
+          var sum = 0;
+          final originalLength = originalKeyData.length;
+          RangeError.checkValueInInterval(lengths.length, 1, originalLength);
+          for (final length in lengths) {
+            RangeError.checkValueInInterval(length, 1, originalLength);
+            RangeError.checkValidRange(sum, sum + length, originalLength);
+            sum += length;
+          }
+          RangeError.checkValueInInterval(length, 1, originalLength);
```

Comment:
I am not sure what this check does? `length` in this context is the same as `originalLength`, so this will always succeed.

Reply:
Looks like a copy-paste error. I deleted it and retested. Everything passed for me. Pushed the update. Thx!